pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/dictionary/MutableDictionary.java

```diff
@@ -49,4 +51,38 @@ public abstract boolean inRange(@Nonnull String lower, @Nonnull String upper, in
 
   public abstract boolean isEmpty();
 
+  /**
+   * Helper method to identify if given (Comparable) value is in provided range.
+   *
+   * @param lower Lower value of range
+   * @param upper Upper value of range
+   * @param includeLower Include lower value in range
+   * @param includeUpper Include upper value in range
+   * @param value Value to compare
+   * @param <T> Extends Comparable
+   * @return True if value in range, false otherwise.
+   */
+  protected <T extends Comparable<T>> boolean valueInRange(@Nonnull T lower, @Nonnull T upper, boolean includeLower,
```

Comment:
This one will cause performance degrade for primitive type (the **inRange()** method is called for all the values in the dictionary)

Reply:
Yes, seems like for mutable dictionaries, we read all values of dictionary for each range query. Reverting back the change for primitive data types, keeping it for String.